src/Language/DifferentialDatalog/Expr.hs

```diff
@@ -527,6 +528,14 @@ exprIsPure d ctx e | isNothing funcs = False
                    | otherwise = all (\f -> not $ funcGetSideEffectAttr d f) $ fromJust funcs
     where funcs = exprFuncsRec d ctx e
 
+-- | Expression can be evaluated statically.
```

Comment:
aren't literals in this category too? or constant expressions?

Reply:
For now we only convert function calls to statics.  Most literals and constant expressions are statically evaluated anyway by the Rust compiler.